.github/workflows/ci.yml

```diff
@@ -6,7 +6,7 @@ on:
       - master
     tags:
       - '[0-9]+\.[0-9]+\.[0-9]+'
-  pull_request:
+  pull_request_target: # forks don't have access to secrets if we use `pull_request`, which is required for codecov
     branches:
       - master
 
```

Comment:
I was wondering if we could / should centralise this condition since is used multiple times?

inside `env:` we could add a new line:
```yml
RUN_CI_CONDITION: ${{ github.event_name == 'push' || !github.event.pull_request.head.repo.fork || (github.event.pull_request.head.repo.fork && contains(github.event.pull_request.labels.*.name, 'run ci')) }}
```

and then use it below:

```yml
if: env.RUN_CI_CONDITION
```

does this makes sense?

Reply:
Thanks for your suggestion! 

I agree that we should try to avoid this duplication, and I think this could work but we would probably need to evaluate the env var via `if: {{ env.RUN_CI_CONDITION }}` so that it would coerce to an actual Boolean value.

However, I am not sure if this approach is a good idea because env vars can be overridden (e.g. by writing to `$GITHUB_ENV`), which might be used to circumvent the _protection_ that the `if` condition is supposed to do.

I would be more comfortable if there was some sort of function/macro that we could define which would be evaluated in each `if`, preventing the actual flag value from being tampered with.

Do you agree?

Reply:
Seems good to me 👍🏻